### PR TITLE
Script hooks

### LINKED ISSUE #1: Add support for custom `make` targets
As discussed in https://github.com/osrf/sdformat/pull/266, osrf/sdformat would like to run `make sdf_description` during its CI, but this action doesn't yet support custom `make` targets.

### LINKED ISSUE #2: Add support for custom hooks after install
As discussed in https://github.com/osrf/sdformat/pull/266, osrf/sdformat has example code that it would like to test in CI by compiling after installing the library, but it doesn't yet have a gtest executable for testing them. If we added a custom hook for a script to be run after installation, we could potentially support that use case.

### PR DESCRIPTION
**Pull request by chapulina**
Closes #1 and also closes #2.

Two of the hooks are being used on https://github.com/osrf/sdformat/pull/266

### CODE REVIEW

**Comment by scpeters** [comment]
instead of passing the names of these scripts as variables, what if we just hard-coded the names of expected scripts (like `.github/ci-bionic/after_make.sh`) and then check if that file exists and try to call it if so

I think it's slightly simpler that way

File: entrypoint.sh [hunk 1]
```diff
@@ -36,9 +39,30 @@ apt -y install $APT_DEPENDENCIES
...
+
 make
 
+echo "SCRIPT_AFTER_MAKE"
+if [ ! -z "$SCRIPT_AFTER_MAKE" ] ; then
+  bash $SCRIPT_AFTER_MAKE
+fi
+
 export CTEST_OUTPUT_ON_FAILURE=1
 make test
```
Comment (scpeters) [inline entrypoint.sh:67]: I want to run `make test` before `make install`, so I'd want an extra step after `make test` for `make install` and examples compilation

File: entrypoint.sh [hunk 2]
```diff
@@ -36,9 +39,30 @@ apt -y install $APT_DEPENDENCIES
...
+echo "SCRIPT_BETWEEN_CMAKE_MAKE"
+if [ ! -z "$SCRIPT_BETWEEN_CMAKE_MAKE" ] ; then
+  bash $SCRIPT_BETWEEN_CMAKE_MAKE
+fi
+
 make
 
+echo "SCRIPT_AFTER_MAKE"
+if [ ! -z "$SCRIPT_AFTER_MAKE" ] ; then
+  bash $SCRIPT_AFTER_MAKE
```
[the remark this answers is not in the document]
Comment (chapulina) [reply entrypoint.sh:64]: I only saw this after merging some "after make" scripts for other libs. I updated the documentation to explain it's before `make test`. What do you think? 27dc5baefafb196fc2fd48353f4e6c03e22e3cfb

File: entrypoint.sh (hunk 1 above)
Comment (chapulina) [reply entrypoint.sh:68]: 27dc5baefafb196fc2fd48353f4e6c03e22e3cfb

File: entrypoint.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (scpeters) [reply entrypoint.sh:64]: yeah that's fine

**Comment by chapulina** [comment]
> instead of passing the names of these scripts as variables, what if we just hard-coded the names of expected scripts

I just noticed this comment. Yeah I think it's a good idea, but now a lot of PRs using that have been merged. So I'll just leave it as is for now and ticketed #8 so we can keep it in mind for the next pass.

**Pull request merged**